core/chain_indexer.go

```diff
@@ -376,8 +380,94 @@ func (c *ChainIndexer) PruneOldBlockData(blockHeight uint64) {
 	rawdb.DeletePendingEtxsRollup(c.chainDb, blockHash)
 	rawdb.DeleteManifest(c.chainDb, blockHash)
 	rawdb.DeletePbCacheBody(c.chainDb, blockHash)
-	rawdb.DeleteSpentUTXOs(c.chainDb, blockHash)
+	createdUtxos, _ := rawdb.ReadCreatedUTXOKeys(c.chainDb, blockHash)
+	createdUtxosToKeep := make([][]byte, 0, len(createdUtxos))
+	for _, key := range createdUtxos {
+		data, _ := c.chainDb.Get(key)
+		if len(data) == 0 {
```

Comment:
how is it possible to have a nonexistent UTXO in the list of `createdUtxos` for a given block? Wouldn't this be a pretty bad bug?

Reply:
Not a bug. If a user spends a UTXO created in block A, the UTXO key will still be in the createdUtxos but won't be on disk.

Reply:
ahh, checks out